paimon-flink/paimon-flink-common/src/main/java/org/apache/paimon/flink/sink/FlinkTableSink.java

```diff
@@ -135,4 +233,50 @@ public RowLevelDeleteInfo applyRowLevelDelete(
                             table.getClass().getName()));
         }
     }
+
+    private boolean shouldPushdownDeleteFilter(Predicate predicate, int predicateSize) {
```

Comment:
Use `FunctionVisitor` to visit predicate?

Reply:
hi, @JingsongLi , is the visitor too complicated for this to implement all the method visitXXX of interface FunctionVisitor?

When the filter is `where id=1 or id=2 or id=3` or `where id in (1,2,3)`, it will be converted to 
```sql
               compound(OR)
                   /     \
         compound(OR)     leaf(Equal, id, 1)
         /        \
leaf(Equal,id,2)   leaf(Equal,id,3)
```

We just need to check all the compoundFilter's func is OR and all the leafFilter's func is Equal and field is in primary keys.

Reply:
We can just throw exception if encounter an unsupported predicate.
And catch the specific exception outer.

This is a common code style in flink-calcite.

Reply:
And we can support `IN` too.